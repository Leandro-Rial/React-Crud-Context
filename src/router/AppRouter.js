import React from "react";
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import AddUser from '../components/AddUser';
import ContextScreen from '../components/ContextScreen';
import EditUser from '../components/EditUser';

const AppRouter = () => {
  return (
    <div style={{ maxWidth: "50rem", margin: "4rem auto" }}>
      <Router>
        <Switch>
          <Route exact path="/" component={ContextScreen} />
          <Route exact path="/add" component={AddUser} />
          <Route exact path="/edit/:id" component={EditUser} />
        </Switch>
      </Router>
    </div>
  );
};

export default AppRouter;
