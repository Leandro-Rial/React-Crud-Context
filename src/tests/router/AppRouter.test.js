import { shallow } from "enzyme";
import AppRouter from "../../router/AppRouter";

describe('Pruebas en AppRouter', () => {
    test('debe de mostar el AppRouter correctamente', () => {
        const wrapper = shallow(
            <AppRouter />
        )

        expect(wrapper).toMatchSnapshot()
    });
});