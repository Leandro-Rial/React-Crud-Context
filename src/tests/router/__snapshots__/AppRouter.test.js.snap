// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AppRouter debe de mostar el AppRouter correctamente 1`] = `
<div
  style={
    Object {
      "margin": "4rem auto",
      "maxWidth": "50rem",
    }
  }
>
  <BrowserRouter>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/add"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/edit/:id"
      />
    </Switch>
  </BrowserRouter>
</div>
`;
